staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation.go

```diff
@@ -86,11 +113,21 @@ func validateStructuralInvariants(s *Structural, lvl level, fldPath *field.Path)
 	if s.Type == "array" && s.Items == nil {
 		allErrs = append(allErrs, field.Required(fldPath.Child("items"), "must be specified"))
 	}
-	allErrs = append(allErrs, validateStructuralInvariants(s.Items, itemLevel, fldPath.Child("items"))...)
+	allErrs = append(allErrs, validateStructuralInvariants(s.Items, itemLevel, fldPath.Child("items"), opts)...)
 
 	for k, v := range s.Properties {
-		allErrs = append(allErrs, validateStructuralInvariants(&v, fieldLevel, fldPath.Child("properties").Key(k))...)
+		allErrs = append(allErrs, validateStructuralInvariants(&v, fieldLevel, fldPath.Child("properties").Key(k), opts)...)
+	}
+
+	if s.AdditionalProperties != nil {
```

Comment:
To see whether we are on the same page:

We do forbid having `properties` and `additionalProperties` in parallel?

And the important question: this schema:
```yaml
properties:
  a:
    type: string
allOf:
  additionalProperties:
    pattern: .*
```
will prune `b` from `{a:"foo",b:"bar"}`?

Reply:
Should be the case as the pruning algorithm is not changed, and it only follows the structural skeleton of the schema.

Reply:
Yeah, `properties` and `additionalProperties` are still mutually exclusive.

Pruning behavior is unchanged. The intention is to allow `additionalProperties` to be placed in NestedValueValidations of a schema defined structurally by `properties`